core/tests/test_apps.bitcoin.signtx_decred.py

```diff
@@ -0,0 +1,368 @@
+                    request_index=0,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevInput(tx=TxAckPrevInputWrapper(input=pinp1)),
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(
+                    request_index=1,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevInput(tx=TxAckPrevInputWrapper(input=pinp2)),
+            TxRequest(
+                request_type=TXOUTPUT,
+                details=TxRequestDetailsType(
+                    request_index=0,
+                    tx_hash=unhexlify(
+                        "df8f9cf58455e8aa22d7f7be09d7877f7a0a698da7695152374c057a3047c24a"
+                    ),
+                ),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckPrevOutput(tx=TxAckPrevOutputWrapper(output=pout1)),
+            TxRequest(
+                request_type=TXINPUT,
+                details=TxRequestDetailsType(request_index=0, tx_hash=None),
+                serialized=EMPTY_SERIALIZED,
+            ),
+            TxAckInput(tx=TxAckInputWrapper(input=inp1)),
+            TxRequest(
+                request_type=TXFINISHED,
+                details=TxRequestDetailsType(request_index=None, tx_hash=None),
+                serialized=TxRequestSerializedType(
+                    signature_index=0,
+                    signature=unhexlify(
+                        "3044022078a5c388838796562eb9dad176b00e6d9425bc360083f633a14948685ca8a5ce02202a1b49cd44104a9d40aee8f988281a8aac94a497b5bc7337c77cc7ddbab16f23"
+                    ),
+                ),
+            ),
+        ]
+
+        seed = bip39.seed(
+            "alcohol woman abuse must during monitor noble actual mixed trade anger aisle",
+            "",
+        )
+        ns = get_schemas_for_coin(coin_decred)
+        keychain = Keychain(seed, coin_decred.curve_name, ns)
+        approver = BasicApprover(tx, coin_decred)
+        signer = decred.Decred(tx, keychain, coin_decred, approver).signer()
+
+        for request, response in chunks(messages, 2):
+            res = signer.send(request)
+            if isinstance(res, tuple):
+                _, res = res
+
+            if isinstance(res, TxRequest):
+                res.serialized.serialized_tx = bytearray(res.serialized.serialized_tx)
+                if not res.serialized.serialized_tx:
+                    res.serialized.serialized_tx = bytearray(0)
+                if not response.serialized.serialized_tx:
+                    response.serialized.serialized_tx = bytearray(0)
```

Comment:
I think something very wierd is going on here. I get strange results if I do not ensure that res.serialized.serialized_tx is a bytearray. I will look into it, but if you guys have any idea I would appreciate some advice.

Reply:
Remove this code. It looks like an innocent type cast, but it actually breaks bitcoin.py. We have a preallocated bytearray there which we keep reusing to reduce the number of memory allocations, and this causes it to be replaced. For the correct solution see comments above.